recipes/gdal/post_3.5.0/conanfile.py

```diff
@@ -0,0 +1,490 @@
+        cmake.definitions["GDAL_USE_JSONC_INTERNAL"] = False
+        cmake.definitions["GDAL_USE_JPEG_INTERNAL"] = False
+        cmake.definitions["GDAL_USE_JPEG12_INTERNAL"] = False
+        cmake.definitions["GDAL_USE_TIFF_INTERNAL"] = False
+        cmake.definitions["GDAL_USE_GEOTIFF_INTERNAL"] = False
+        cmake.definitions["GDAL_USE_GIF_INTERNAL"] = False
+        cmake.definitions["GDAL_USE_PNG_INTERNAL"] = False
+
+        cmake.definitions["GDAL_USE_LERC_INTERNAL"] = True
+        cmake.definitions["GDAL_USE_SHAPELIB_INTERNAL"] = True
+
+        cmake.definitions["BUILD_APPS"] = self.options.tools
+        del cmake.definitions["CMAKE_MODULE_PATH"]
+
+        cmake.definitions["SQLite3_HAS_COLUMN_METADATA"] = \
+            self.options["sqlite3"].enable_column_metadata
+
+        cmake.definitions["SQLite3_HAS_RTREE"] = self.options[
+            "sqlite3"].enable_rtree
+
+        if self.options.with_hdf5:
+            cmake.definitions["HDF5_C_LIBRARIES"] = "HDF5::C"
+            cmake.definitions["HDF5_BUILD_SHARED_LIBS"] = self.options["hdf5"].shared
+
+        for dep in self.gdal_deps:
+            upper = dep["dep"].upper()
+            enabled = "require" in dep and getattr(self.options,
+                                                   _option_name(dep))
+            cmake.definitions["GDAL_USE_" + upper] = enabled
+            if enabled:
+                cmake.definitions["GDAL_CONAN_PACKAGE_FOR_" + upper] = \
+                       _strip_version(dep["require"])
+            else:
+                cmake.definitions[dep["dep"] + "_FOUND"] = False
+
+        for k, v in cmake.definitions.items():
+            print(k, " = ", v)
+
+        cmake.configure()
+        return cmake
+
+    def build(self):
+        apply_conandata_patches(self)
+        cmake = self._configure_cmake()
+
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE.TXT", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "lib/pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "lib/cmake"))
+
+    def package_info(self):
+
+        # It would be better to use GDAL-targets.cmake generated by cmake,
+        # but I do not know how to do that.
+        self.cpp_info.set_property("cmake_file_name", "GDAL")
+        self.cpp_info.set_property("cmake_target_name", "GDAL::GDAL")
+        self.cpp_info.set_property("cmake_find_mode", "both")
+        self.cpp_info.set_property("pkg_config_name", "gdal")
+
+        self.cpp_info.names["cmake_find_package"] = "GDAL"
+        self.cpp_info.names["cmake_find_package_multi"] = "GDAL"
+        self.cpp_info.filenames["cmake_find_package"] = "GDAL"
+        self.cpp_info.filenames["cmake_find_package_multi"] = "GDAL"
+
+        self.cpp_info.libs = tools.collect_libs(self)
```

Comment:
Is lib name predictable?

Reply:
yes, `add_library("GDAL"...)`. Removed tools.collect_libs.